react/src/components/appointments.js

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+
+import { setAppointmentsToActiveComponent, logOutUser } from '../actions/index';
+import { getAllTheAppointments } from '../utilities/api-calls';
+
+import '../css/appointments.css';
+
+const Appointments = props => {
+  const [appointments, setAppointments] = useState([]);
+  const { token, setAppointmentsToActiveComponent } = props;
+
+  const getAppointments = async () => {
+    setAppointments(await getAllTheAppointments(token));
+  };
+
+  useEffect(() => {
+    getAppointments();
+    setAppointmentsToActiveComponent();
+  }, []);
+
+  let keyIndex = 0;
+
+  return (
+    <div className="appointments-wrapper">
+      {appointments.map(appointment => (
+        <div className="appointment" key={`appointment-${keyIndex += 1}`}>
```

Comment:
Does the api not include the `id` of an appointment? If so, make use of it as the key. Otherwise, use the index of the current appointment item.

```
{!!appointments && appointments.map((appointment, index) => (
  <div key={`${index}`} className="appointment">
```

Reply:
Well, again, I already explained that in the reply email to Israel the first time. Like I have to be knowing the fact, that he is the middle man and even then, that the reply isn't forwarded or something. It looks like that project, in the end, gets reviewed a year or something. On that page, if I remember correctly, the warning popped, that the ids were overlapping between two components, data structure ids are not global between containers - Like honestly, what is the overlaying problem even here, to point it out? It doesn't affect in any way, how the program works, only makes it sure, that the ids won't overlap.